src/bacnet/basic/object/device.c

```diff
@@ -1998,14 +1998,17 @@ bool Device_Write_Property(BACNET_WRITE_PROPERTY_DATA *wp_data)
                 }
                 if (status) {
                     Device_Write_Property_Store(wp_data);
+                } else {
+                    wp_data->error_class = ERROR_CLASS_PROPERTY;
+                    wp_data->error_code = ERROR_CODE_INVALID_DATA_TYPE;
                 }
             } else {
                 wp_data->error_class = ERROR_CLASS_PROPERTY;
                 wp_data->error_code = ERROR_CODE_WRITE_ACCESS_DENIED;
             }
         } else {
             wp_data->error_class = ERROR_CLASS_OBJECT;
-            wp_data->error_code = ERROR_CODE_UNKNOWN_OBJECT;
+            wp_data->error_code = ERROR_CODE_UNSUPPORTED_OBJECT_TYPE;
```

Comment:
no! this is not correct : when NOT pObject->Object_Valid_Instance &&
pObject->Object_Valid_Instance(wp_data->object_instance))  then  ERROR_CODE_UNKNOWN_OBJECT .